shesha-core/src/Shesha.Core/Migrations/M20241203150000.cs

```diff
@@ -0,0 +1,193 @@
+                  .WithColumn(DatabaseConsts.ExtSysLastSyncDate).AsDateTime().Nullable()
+                  .WithColumn(DatabaseConsts.ExtSysSource).AsString(50).Nullable()
+                  .WithColumn(DatabaseConsts.ExtSysSyncError).AsStringMax().Nullable()
+                  .WithColumn(DatabaseConsts.ExtSysSyncStatusLkp).AsInt32().Nullable();
+
+            Alter.Table("Core_NewNotificationMessages")
+                  .AddTenantIdColumnAsNullable();
+
+            Execute.Sql(@"
+                INSERT INTO Core_NewNotifications (Id, CreationTime, Name, ToPersonId, FromPersonId, TriggeringEntityId, TriggeringEntityClassName, TriggeringEntityDisplayName) 
+                SELECT 
+	                NEWID(),
+	                nm.CreationTime,
+                    n.Name,
+                    nm.RecipientId,
+                    nm.SenderId,
+                    nm.SourceEntityId,
+                    nm.SourceEntityClassName,
+                    nm.SourceEntityDisplayName
+                FROM Core_Notifications n
+                LEFT JOIN Core_NotificationMessages nm ON n.Id = nm.NotificationId
+                WHERE nm.IsDeleted = 0 AND n.IsDeleted = 0;
+            ");
+
+            Execute.Sql(@"
+                INSERT INTO Core_NewNotificationMessages (Id, CreationTime, PartOfId, ChannelId, RecipientText, Subject, Message,RetryCount, DirectionLkp, ReadStatusLkp, FirstDateRead,DateSent, ErrorMessage, StatusLkp)
+                SELECT 
+	                Id,
+	                CreationTime,
+	                NotificationId,
+                    CASE 
+                        WHEN SendTypeLkp = 1 THEN 
+                            (SELECT Id FROM Core_NotificationChannelConfigs ncc WHERE ncc.Core_SenderTypeName = 'EmailChannelSender')
+                        WHEN SendTypeLkp = 2 THEN 
+                            (SELECT Id FROM Core_NotificationChannelConfigs ncc WHERE ncc.Core_SenderTypeName = 'SmsChannelSender')
+                        ELSE NULL
+                    END,
+                    RecipientText,
+                    Subject,
+                    Body,
+                    TryCount,
+                    DirectionLkp,
+                    CASE WHEN Opened = 1 THEN 0 ELSE 1 END,
+                    LastOpened,
+                    SendDate,
+                    ErrorMessage,
+                    StatusLkp
+                FROM Core_NotificationMessages
+                WHERE IsDeleted = 0;
+            ");
+
+
+            Execute.Sql(@"
+                INSERT INTO Core_NewNotificationTemplates (Id, TitleTemplate, BodyTemplate, MessageFormatLkp)
+                SELECT 
+	                Id,
+	                Subject,
+	                Body,
+	                BodyFormatLkp
+                FROM Core_NotificationTemplates
+                WHERE IsDeleted = 0;
+            ");
+
+
+            Rename.Table("Core_NotificationMessages").To("Core_OldNotificationMessages");
```

Comment:
Please rename tables before creation of new ones, otherwise some auto-generated names (FKs, indexes etc.) will contain wrong table names with `New` part